packages/Renaming/src/Rector/Class_/RenameClassRector.php

```diff
@@ -323,4 +327,17 @@ private function refactorPhpDoc(Node $node): void
 
         $this->phpDocClassRenamer->changeTypeInAnnotationTypes($node, $this->oldToNewClasses);
     }
+
+    private function ensureClassWillNotBeDuplicate(string $newName, string $oldName): void
+    {
+        if (! ClassExistenceStaticHelper::doesClassLikeExist($newName)) {
+            return;
+        }
+
+        throw new InvalidPhpCodeException(sprintf(
+            'Renaming class "%s" to "%s" would create a duplicated class/interface/trait and cause PHP code to be invalid',
+            $oldName,
+            $newName
```

Comment:
Todo: show exact location of the existing class file, so people can investigate better